### PR TITLE
#22 Update HTTP client configuration to include maxTTL for per route

### PR DESCRIPTION
**Pull request by Monoradioactivo**
### Checklist
🚨 Please review this repository's [contribution guidelines](./CONTRIBUTING.md).

- [ x] I've read and agree to the project's contribution guidelines.
- [ x] I'm requesting to **pull a topic/feature/bugfix branch**.
- [ x] I checked that my code additions will pass code linting checks and unit tests.
- [ x] I updated unit and integration tests (if applicable).
- [x ] I'm ready to notify the team of this contribution.

### Description
The provided modifications introduce a capability to specify a custom TTL for connections based on the specific routes or hostnames they target. Instead of a blanket TTL for all connections, the system can now differentiate and apply unique TTLs to different routes. This offers a more flexible configuration, allowing for tailored connection management based on the characteristics or requirements of each individual route. The changes in the test class ensure that this new functionality is validated, ensuring that connections to specified routes do adhere to the set TTLs

[Link to JIRA]

Thank you!

### CODE REVIEW

**Review by aditi2205 — APPROVED** [review]

File: src/main/java/com/intuit/springwebclient/config/HttpClientConfig.java [hunk 1]
```diff
@@ -1,9 +1,21 @@
 package com.intuit.springwebclient.config;
 
+import java.util.HashMap;
+import java.util.Map;
 import lombok.Data;
 
 @Data
 public class HttpClientConfig {
-    private Integer connectTimeoutMillis = 30000;
```
Comment (aditi2205) [inline src/main/java/com/intuit/springwebclient/config/HttpClientConfig.java:7]: unchanged file, keep as is

File: src/main/java/com/intuit/springwebclient/config/WebClientConfiguration.java [hunk 2]
```diff
@@ -21,31 +23,50 @@
 @Configuration
 @AllArgsConstructor
 public class WebClientConfiguration {
-    private final SpringWebClientConfig webClientConfiguration;
```
Comment (aditi2205) [inline src/main/java/com/intuit/springwebclient/config/WebClientConfiguration.java:24]: unnecessary deletions

**Comment by Monoradioactivo** [comment]
Would you mind adding the hacktoberfest topic to the repository, or add the hacktoberfest-accepted label please?

**Comment by raghav-agarwal** [comment]
Stale pull request message

**Pull request closed without merging**